### PR TITLE
Add find bugs plugin to ob1k compilation

### PR DESCRIPTION
**Pull request by lifey**

### CODE REVIEW

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java [hunk 1]
```diff
@@ -56,14 +60,16 @@ public Check getCheck() {
...
+      interval = null;
+      timeout = null;
     }
 
     public Check(final String url, final int intervalSec) {
-      this.http = Preconditions.checkNotNull(url, "url must not be null");
+      http = Preconditions.checkNotNull(url, "url must not be null");
       Preconditions.checkArgument(0 < intervalSec, "intervalSec must be greater than zero");
-      this.interval = intervalSec + "s";
-      this.timeout = interval;
```
Comment (marenzo) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: findbugs doesn't like referencing to fields by "this"?

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java [hunk 2]
```diff
@@ -18,7 +18,11 @@
 
   // I need to make some marshalers happy ;)
   ServiceRegistration() {
-    this(null, 0, null, null, null);
+    name = null;
```
Comment (marenzo) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: meh

File: ob1k-core/src/main/java/com/outbrain/ob1k/common/marshalling/MessagePackRequestMarshaller.java [hunk 3]
```diff
@@ -263,14 +263,14 @@ public HttpContent marshallResponsePart(final Object res, final HttpResponseStat
       final Template<T> template = (Template<T>) msgPack.lookup(type);
       return template.read(new Converter(msgPack, value), null);
 
-    } else if (Arrays.equals(ChunkHeader.ERROR_HEADER.getBytes(), header)) {
+    } else if (Arrays.equals(ChunkHeader.ERROR_HEADER.getBytes(CharsetUtil.UTF_8), header)) {
 
       final byte[] body = new byte[remaining];
       byteBufferBody.get(body);
-      throw new RuntimeException(new String(body));
+      throw new RuntimeException(new String(body,CharsetUtil.UTF_8));
```
Comment (marenzo) [inline ob1k-core/src/main/java/com/outbrain/ob1k/common/marshalling/MessagePackRequestMarshaller.java]: a bit more padding :)

Comment (langera) [inline ob1k-core/src/main/java/com/outbrain/ob1k/common/marshalling/MessagePackRequestMarshaller.java]: That check can probably be added to the excludes, no?

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java (hunk 2 above)
Comment (langera) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: No Michael. 
If your approach is that you know better and findbugs is the annoying thing warning about meaningless stuff then there's no point in running it. We might as well remove it. 
Its a tool that is there to help you. So "meh" is wrong here.

findbugs is dead right here and points to a real problem in the code. 

There are two options here:
1. If marshalling is not needed then remove it.
2. If it is needed then it shouldn't throw an NPE. 

I think this fix points at option 2 but I would not use null when there's another constructor that checks that the field should never be null. I would use an empty string but I guess that is more style than correctness.

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java (hunk 1 above)
Comment (langera) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: Don't think that's findbugs. 
I like using "this" - it more readbale this way.

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java (hunk 2 above)
Comment (eranharel) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: The problem was that MsgPack doesn't like the no-empty c'tor even though we don't use MsgPack here at all.
We use JSON here, so this is just a hack basically. The fix is similar to the c'tor, so whatever.

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java (hunk 1 above)
Comment (eranharel) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: You like using this, and you removed it?
The reason I use this.xxx=xxx in c'tors is that it prevents bugs like setting a variable to itself.

Comment (langera) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: emm... no. 

I like using "this"

and Haim removed it.

File: ob1k-core/src/main/java/com/outbrain/ob1k/common/marshalling/MessagePackRequestMarshaller.java (hunk 3 above)
Comment (lifey) [inline ob1k-core/src/main/java/com/outbrain/ob1k/common/marshalling/MessagePackRequestMarshaller.java]: Yes but why ? There are a few changes needed to be made, some places have the encoding and we do need to think about it as it is an open source

File: ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java (hunk 1 above)
Comment (lifey) [inline ob1k-consul/src/main/java/com/outbrain/ob1k/consul/ServiceRegistration.java]: I will add it back.

**Pull request merged**